OpenRA.Mods.Common/Traits/Render/WithGateSpriteBody.cs

```diff
@@ -18,7 +18,7 @@
 
 namespace OpenRA.Mods.Common.Traits.Render
 {
-	class WithGateSpriteBodyInfo : WithSpriteBodyInfo, IWallConnectorInfo, Requires<GateInfo>
+	class WithGateSpriteBodyInfo : WithSpriteBodyInfo, IWallConnectorInfo, Requires<BuildingInfo>, Requires<GateInfo>
```

Comment:
I am not against it, but we should be clear about whether we want redundant `requires` (`GateInfo` already requires `BuildingInfo`).

Reply:
Relying on transitive dependencies is a massive code smell.  This trait requires `Building` to work correctly, so should declare that dependency independently from what any other trait does.

But, saying that, IMO it would be cleaner to expose the footprint from `Gate` so that this trait doesn't need to reference `Building` at all.

Reply:
Relying on transitive dependencies is a massive code smell.  This trait requires `Building` to work correctly, so should declare that dependency independently from what any other trait does.

But, saying that, IMO it would be cleaner to expose the footprint from `Gate` so that this trait doesn't need to reference `Building` at all.